hivemq-edge/src/main/java/com/hivemq/configuration/entity/combining/DataCombiningSourcesEntity.java

```diff
@@ -16,47 +16,43 @@
 package com.hivemq.configuration.entity.combining;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.hivemq.combining.model.PrimaryType;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import javax.xml.bind.annotation.XmlElement;
 import javax.xml.bind.annotation.XmlElementWrapper;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Objects;
 
 
 public class DataCombiningSourcesEntity {
 
-    @JsonProperty("primaryName")
-    @XmlElement(name = "primary-name")
-    private @NotNull String primaryName;
-
-    @JsonProperty("primaryType")
-    @XmlElement(name = "primary-type")
-    private @NotNull PrimaryType primaryType;
+    @JsonProperty("primaryReference")
+    @XmlElement(name = "primaryReference")
```

Comment:
Please mind the XML notation.
JSON is camelcase, XML Kebap-case:
JSON: primaryReference
XM: primary-reference